phase/generator_test.go

```diff
@@ -176,465 +174,696 @@ func testGenerator(t *testing.T, when spec.G, it spec.S) {
 						dirStore.EXPECT().LookupExt("A", "v1").Return(&extA, nil)
-						executor.EXPECT().Generate(extA, gomock.Any(), gomock.Any()).Return(buildpack.GenerateOutputs{
-							Dockerfiles: tc.aDockerfiles,
-						}, nil)
+						executor.EXPECT().Generate(extA, gomock.Any(), gomock.Any()).Return(buildpack.GenerateOutputs{}, errors.New("some error"))
 
-						// mock generate for extension B
-						dirStore.EXPECT().LookupExt("ext/B", "v2").Return(&extB, nil)
-						executor.EXPECT().Generate(extB, gomock.Any(), gomock.Any()).Return(buildpack.GenerateOutputs{
-							Dockerfiles: tc.bDockerfiles,
-						}, nil)
-
-						// do generate
-						result, err := generator.Generate()
-						if err == nil {
-							h.AssertEq(t, result.AnalyzedMD.RunImage.Image, tc.expectedRunImageImage)
-							h.AssertEq(t, result.AnalyzedMD.RunImage.Reference, tc.expectedRunImageReference)
-							h.AssertEq(t, result.AnalyzedMD.RunImage.Extend, tc.expectedRunImageExtend)
-						} else {
-							t.Log(err)
-							h.AssertError(t, err, tc.expectedErr)
+						if _, err := generator.Generate(); err == nil {
+							t.Fatal("Expected error.\n")
+						} else if !strings.Contains(err.Error(), "some error") {
+							t.Fatalf("Incorrect error: %s\n", err)
 						}
+					})
+				})
+
+				when("later extension fails", func() {
+					it("errors", func() {
+						extA := buildpack.ExtDescriptor{Extension: buildpack.ExtInfo{BaseInfo: buildpack.BaseInfo{ID: "A", Version: "v1"}}}
+						dirStore.EXPECT().LookupExt("A", "v1").Return(&extA, nil)
+						executor.EXPECT().Generate(extA, gomock.Any(), gomock.Any()).Return(buildpack.GenerateOutputs{}, nil)
+						extB := buildpack.ExtDescriptor{Extension: buildpack.ExtInfo{BaseInfo: buildpack.BaseInfo{ID: "ext/B", Version: "v1"}}}
+						dirStore.EXPECT().LookupExt("ext/B", "v2").Return(&extB, nil)
+						executor.EXPECT().Generate(extB, gomock.Any(), gomock.Any()).Return(buildpack.GenerateOutputs{}, errors.New("some error"))
 
-						if tc.assertAfter != nil {
-							tc.assertAfter()
+						if _, err := generator.Generate(); err == nil {
+							t.Fatal("Expected error.\n")
+						} else if !strings.Contains(err.Error(), "some error") {
+							t.Fatalf("Incorrect error: %s\n", err)
 						}
 					})
 				})
-			}
+			})
 		})
 
-		when("extension generate failed", func() {
-			when("first extension fails", func() {
-				it("errors", func() {
-					extA := buildpack.ExtDescriptor{Extension: buildpack.ExtInfo{BaseInfo: buildpack.BaseInfo{ID: "A", Version: "v1"}}}
-					dirStore.EXPECT().LookupExt("A", "v1").Return(&extA, nil)
-					executor.EXPECT().Generate(extA, gomock.Any(), gomock.Any()).Return(buildpack.GenerateOutputs{}, errors.New("some error"))
-
-					if _, err := generator.Generate(); err == nil {
-						t.Fatal("Expected error.\n")
-					} else if !strings.Contains(err.Error(), "some error") {
-						t.Fatalf("Incorrect error: %s\n", err)
-					}
-				})
+		when("using the platform API 0.13", func() {
```

Comment:
I'm not sure it's worth having a whole separate block for the 0.13 tests - it looks like if we run the 0.12 block with `generator.PlatformAPI = api.MustParse("0.13")` the only two tests that fail are:
* `it("copies Dockerfiles and extend-config.toml files to the correct locations", func() {`
* `descCondition: "a run.Dockerfile declares a new base image (only) and no run.Dockerfiles follow",
						descResult:    "returns the referenced base image, and sets extend to false",`

It might make our future lives easier to just conditionally set a variable for "dockerfile parent dir" depending on the Platform API and use that in the tests